src/main/java/org/mastodon/mamut/util/LineageTreeUtils.java

```diff
@@ -22,27 +52,264 @@ public class LineageTreeUtils {
 	 */
-	public static void callDepthFirst(@Nonnull ModelBranchGraph branchGraph, @Nonnull Consumer<BranchSpot> action) {
-		DepthFirstSearch<BranchSpot, BranchLink> search = new DepthFirstSearch<>(branchGraph, GraphSearch.SearchDirection.DIRECTED);
-		search.setTraversalListener(new SearchListener<BranchSpot, BranchLink, DepthFirstSearch<BranchSpot, BranchLink>>() {
+	public static void callDepthFirst( @Nonnull ModelBranchGraph branchGraph, @Nonnull Consumer< BranchSpot > action )
+	{
+		DepthFirstSearch< BranchSpot, BranchLink > search = new DepthFirstSearch<>( branchGraph, GraphSearch.SearchDirection.DIRECTED );
+		search.setTraversalListener( new SearchListener< BranchSpot, BranchLink, DepthFirstSearch< BranchSpot, BranchLink > >()
+		{
 			@Override
-			public void processVertexLate(BranchSpot vertex, DepthFirstSearch<BranchSpot, BranchLink> search) {
-				action.accept(vertex);
+			public void processVertexLate( BranchSpot vertex, DepthFirstSearch< BranchSpot, BranchLink > search )
+			{
+				action.accept( vertex );
 			}
 
 			@Override
 			public void processVertexEarly( BranchSpot vertex, DepthFirstSearch< BranchSpot, BranchLink > search )
-			{}
+			{
+				throw new UnsupportedOperationException();
+			}
 
 			@Override
 			public void processEdge( BranchLink edge, BranchSpot from, BranchSpot to, DepthFirstSearch< BranchSpot, BranchLink > search )
-			{}
+			{
+				throw new UnsupportedOperationException();
+			}
 
 			@Override
 			public void crossComponent( BranchSpot from, BranchSpot to, DepthFirstSearch< BranchSpot, BranchLink > search )
-			{}
+			{
+				throw new UnsupportedOperationException();
+			}
 		} );
 		final RefSet< BranchSpot > roots = RootFinder.getRoots( branchGraph );
 		roots.forEach( search::start );
 	}
+
+	private static final String GRAPHML_VERTEX_LABEL_ATTRIBUTE_NAME = "branchSpotName";
+
+	private static final String GRAPHML_VERTEX_LIFESPAN_ATTRIBUTE_NAME = "lifespan";
+
+	/**
+	 * Gets the first time point with the given number of spots ({@code numberOfSpots})
+	 * by iterating through the given spatio-temporal index ({@code spotSpatioTemporalIndex})
+	 * from the given minimum time point ({@code minTimePoint}) to the given maximum time point ({@code maxTimePoint}).
+	 *
+	 * @param spotSpatioTemporalIndex the index to search in
+	 * @param minTimePoint the minimum time point to search in (inclusive)
+	 * @param maxTimePoint the maximum time point to search in (inclusive)
+	 * @param numberOfSpots the number of spots to search for
+	 * @return the first time point with the given number of spots or -1 if no such time point exists
+	 */
+	public static int getTimePointWithNSpots( @Nonnull SpatioTemporalIndex< Spot > spotSpatioTemporalIndex,
+			int minTimePoint, int maxTimePoint, int numberOfSpots )
+	{
+		for ( int timePoint = minTimePoint; timePoint <= maxTimePoint;
+				timePoint++ )
+			if ( spotSpatioTemporalIndex.getSpatialIndex( timePoint ).size() == numberOfSpots )
```

Comment:
The number of timepoints might jump from 99 to 101 if two cells divide for t=99. It's probably much better if you implement a method that return the first time point with more than the requested number of cells.

Reply:
You are right. I can remember that I already changed this, but somehow I pushed the wrong version. I changed it now to equal or more. Would you agree that _equal or more_ is correct? I am asking, since you wrote _more_.